x-pack/plugins/aiops/public/components/change_point_detection/fields_config.tsx

```diff
@@ -0,0 +1,306 @@
+              }}
+              grow={false}
+            >
+              <EuiProgress
+                label={
+                  <FormattedMessage
+                    id="xpack.aiops.changePointDetection.progressBarLabel"
+                    defaultMessage="Fetching change points"
+                  />
+                }
+                value={progress ?? 0}
+                max={100}
+                valueText
+                size="m"
+              />
+              <EuiSpacer size="s" />
+            </EuiFlexItem>
+          </FieldsControls>
+        }
+        extraAction={
+          <EuiButtonIcon
+            disabled={removeDisabled}
+            aria-label="trash"
+            iconType="trash"
+            color="danger"
+            onClick={onRemove}
+          />
+        }
+        paddingSize="s"
+      >
+        <ChangePointResults
+          fieldConfig={fieldConfig}
+          isLoading={annotationsLoading}
+          annotations={annotations}
+          splitFieldCardinality={splitFieldCardinality}
+          onSelectionChange={onSelectionChange}
+        />
+      </EuiAccordion>
+    </EuiPanel>
+  );
+};
+
+interface FieldsControlsProps {
+  fieldConfig: FieldConfig;
+  onChange: (update: FieldConfig) => void;
+}
+
+/**
+ * Renders controls for fields selection and emits updates on change.
+ * @param fieldConfig
+ * @param onChange
+ * @constructor
+ */
+export const FieldsControls: FC<FieldsControlsProps> = ({ fieldConfig, onChange, children }) => {
+  const { splitFieldsOptions } = useChangePointDetectionContext();
+
+  const onChangeFn = useCallback(
+    (field: keyof FieldConfig, value: string) => {
+      const result = { ...fieldConfig, [field]: value };
+      onChange(result);
+    },
+    [onChange, fieldConfig]
+  );
+
+  return (
+    <EuiFlexGroup alignItems={'center'}>
+      <EuiFlexItem grow={false} css={{ width: '200px' }}>
+        <FunctionPicker value={fieldConfig.fn} onChange={(v) => onChangeFn('fn', v)} />
+      </EuiFlexItem>
+      <EuiFlexItem grow={false} css={selectControlCss}>
+        <MetricFieldSelector
+          value={fieldConfig.metricField!}
+          onChange={(v) => onChangeFn('metricField', v)}
+        />
+      </EuiFlexItem>
+      {splitFieldsOptions.length > 0 ? (
+        <EuiFlexItem grow={false} css={selectControlCss}>
+          <SplitFieldSelector
+            value={fieldConfig.splitField}
+            onChange={(v) => onChangeFn('splitField', v!)}
+          />
+        </EuiFlexItem>
+      ) : null}
+
+      {children}
+    </EuiFlexGroup>
+  );
+};
+
+interface ChangePointResultsProps {
+  fieldConfig: FieldConfig;
+  splitFieldCardinality: number | null;
+  isLoading: boolean;
+  annotations: ChangePointAnnotation[];
+  onSelectionChange: (update: SelectedChangePoint[]) => void;
+}
+
+/**
+ * Handles request and rendering results of the change point  with provided config.
+ */
+export const ChangePointResults: FC<ChangePointResultsProps> = ({
```

Comment:
Nit: There are 4 different components in this file, might be good to separate them out into a `fields_config` folder?

Reply:
I couldn't find any recommendation regarding files structure in the [STYLEGUIDE](https://github.com/elastic/kibana/blob/main/STYLEGUIDE.mdx) 🤔  
I don't mind moving components into separate files, but they all used together and I've split child component just for convenience. And it's 306 LOC in total so I assume it's acceptable and easy to navigate.